src/main/java/com/codepine/api/testrail/Request.java

```diff
@@ -114,6 +114,7 @@ public T execute() {
             String url = getUrl();
             HttpURLConnection con = (HttpURLConnection) urlConnectionFactory.getUrlConnection(url);
             con.setRequestMethod(method.name());
+            con.setDoOutput(true);
```

Comment:
This should happen only during POST. So please move it to the line after

`if (method == Method.POST) {`

and remove the one in

`if (content != null) {`

Also, please add a unit test (delete case) to `RequestTest.java`. If needed I can do so.